### DISCUSSION TITLE
HyperHDR process / load / read video stream (from usb grabber or usb cam) has different/wrong colours than in reality

### DISCUSSION THREAD

**Discussion by LEDExperiments** [awawa-dev/HyperHDR#844] [Q&A] [answered]
Hey guys,

first, thanks for the amazing project! Just started with 20.0.0 version of HyperHDR. First, I wondered why I always see different colours from the LEDs compared to the screen. I tried different colour setups (RGB, BGR, GBR etc) but when I watch the video stream in HyperHDR, it was clear why I cannot find the right one: the colours are different then they should be and different from the TV Screen. Please see here:

![PXL_20240502_134945117 MP](https://github.com/awawa-dev/HyperHDR/assets/168914332/7a64e6d2-c1f0-4b6d-a2b3-de76b65835f7)

Moreover, the same happens when I change from the usb grabber to a video cam (attached via usb port to the pi), the colours are shown differently in HyperHDR, too. So it should nothing be specifically cused by the video signal grabber.
![2024-05-04 10_30_09-Window](https://github.com/awawa-dev/HyperHDR/assets/168914332/fc409e20-6d28-4a27-9b7b-e0bb558f6afd)

And here how it looks on the same pi shown in (motioneye):
![2024-05-04 16_49_25-Window](https://github.com/awawa-dev/HyperHDR/assets/168914332/51762da5-d520-459c-bf35-0f3f2580be72)

Any ideas or recommendations what I is wrong or what I can do? Maybe I have to mention that before HyperHDR, I tried hyperion and the colours were right in the video stream, but I switched because of the more possibilities of HyperHDR. I also tried hyperHDR in 19.0.0 but got the same.

My setup in general: 

**Video signal processing:**

Source (Amazon Fire TV Stick) --> HDMI Splitter --> getting two hdmi outputs
1st Output --> TV
2nd Output --> HDMI Input of  USB Video Grabber --> USB port of Raspberry Pi Zero w2

**HyperHDR**
Running on  Raspberry Pi Zero w2 with version 20.0.0
LED Instance: WLED (ESP32 with WS2815 LED stripe mounted on the backside of the tv)
Connection: via wirless LAN (another issue: results in an observable delay)

Thanks for your help!

**Comment by awawa-dev** [comment]
https://github.com/awawa-dev/HyperHDR/discussions/834
https://github.com/awawa-dev/HyperHDR/discussions/797

**Comment by LEDExperiments** [comment, marked as answer]
I used:

 _sudo rpi-update 5fc4f643d2e9c5aa972828705a902d184527ae3f_

as recommended in #848 and it works again :-)

Thanks for help!
